src/main/daml/Daml/Finance/Holding/Account.daml

```diff
@@ -34,26 +40,50 @@ template Account
     signatory custodian, owner
     observer Disclosure.flattenObservers observers
 
-    let accountKey = AccountKey with custodian; owner; id
+    let
+      account = AccountKey with custodian; owner; id
+      transferors = Transferors with instructors = S.singleton owner; approvers = S.singleton owner
 
     interface instance Account.I for Account where
       asDisclosure = toInterface @Disclosure.I this
-      view = Account.View with custodian; id; owner; holdingFactoryCid; description
-      getKey = accountKey
+      view = Account.View with custodian; id; owner; holdingFactoryCid; description; transferors
+      getKey = account
       credit Account.Credit{quantity} =
         exercise holdingFactoryCid HoldingFactory.Create with
           instrument = quantity.unit
-          account = accountKey
+          account
           amount = quantity.amount
           observers = M.empty
-      debit Account.Debit{holdingCid} =
-        exercise holdingFactoryCid HoldingFactory.Remove with actors = fromList [accountKey.custodian, accountKey.owner]; holdingCid
+      debit Account.Debit{holdingCid} = do
+        vHolding <- view <$> fetch holdingCid
+        assertMsg "accounts must match" $ vHolding.account == account
+        exercise holdingFactoryCid HoldingFactory.Remove with actors = S.fromList [account.custodian, account.owner]; holdingCid
+      transfer Account.Transfer{transferableCid; actors} = do
+        let lockableCid = toInterfaceContractId @Lockable.I transferableCid
+        lockable <- fetch lockableCid
+        let
+          vH = view $ toInterface @Base.I lockable
+          vL = view $ toInterface @Lockable.I lockable
+        assertMsg "custodians must be the same" $ vH.account.custodian == custodian
+        assertMsg "actors must contain lockers" $ S.isSubsetOf (getLocker lockable) actors
+        assertMsg "actors must contain the owner (of the provided transferable)" $ S.member (getOwner lockable) actors
+        -- Release all locks
+        transferableCid <- toInterfaceContractId @Base.I <$> optional (pure lockableCid) (releaseAll lockableCid) vL.lock
```

Comment:
Why we want to allow locked assets to be transferred ?

Reply:
as long as the `lockers` agree it is fine, the locks will then be reapplied to the new holding. I haven't actually changed that, it is how it used to be.

Reply:
> as long as the `lockers` agree it is fine, the locks will then be reapplied to the new holding.

I can imagine that there may be overlap between the `lockers` and the `approvers` set or `custodian` which may lead to locked assets being transferred when they shouldn't be...  

> I haven't actually changed that, it is how it used to be.

I presume this would have been separate choices being called as part of some workflow? Ie, check if locked, if locked then unlock -> transfer -> relock.

I'm just not sure we should have the unlock/relock logic sitting inside the `transfer` choice...

Reply:
In the `Transfer` choice of the `Account`, there is a check that the actors contain the lockers (to avoid overlap):
`assertMsg "actors must contain lockers" $ S.isSubsetOf (getLockers lockable) actors`
I will add the same check to the `Transfer` choice of the holding to avoid the overlap:
`assertMsg "actors must contain all lockers" $ S.isSubsetOf lockers actors`
You could argue the authorizers of the choice should know they are also the lockers, but agree that it is better to make this explicit.

Reply:
I think we do want to allow transfer (and split/merge) of a locked asset, as long as it's authorized by the lockers as well. I guess the fact that we have to uncock before debiting, and lock after crediting comes from the fact that create/remove choices on the holding factory are just controlled by [custodian, owner]? If we were to relax these to take `actors` instead we could directly debit the locked holding, and credit again a locked one with the same lockers. Would make this transfer implementation a bit simpler I guess (not having to explicity unlock & lock), but it adds some complexity to the factory. 🤷‍♂️

Reply:
agree, I'll keep it as is for now to avoid adding extra actors